src/test/java/org/isf/pregtreattype/test/Tests.java

```diff
@@ -116,6 +121,141 @@ public void testIoDeletePregnantTreatmentType() throws Exception {
 		assertThat(result).isFalse();
 	}
 
+	@Test
+	public void testMgrGetPregnantTreatmentType() throws Exception {
+		String code = _setupTestPregnantTreatmentType(false);
+		PregnantTreatmentType foundPregnantTreatmentType = pregnantTreatmentTypeIoOperationRepository.findOne(code);
+		ArrayList<PregnantTreatmentType> pregnantTreatmentTypes = pregnantTreatmentTypeBrowserManager.getPregnantTreatmentType();
+
+		for (PregnantTreatmentType pregnantTreatmentType : pregnantTreatmentTypes) {
+			if (pregnantTreatmentType.getCode().equals(code)) {
```

Comment:
What about asserting something for the else branch?

Reply:
I added something but I'm not sure how much value it adds.

Reply:
@emecas what was you thinking like?

Reply:
@emecas +1

Reply:
Initially I just noticed that branch was not being  tested, and was a good spot for exploratory testing.

`isNotEqualTo` makes more sense because new version of the test  has multiple `PregnantTreatmentType `objects in the ArrayList

Also, it totally depends how you write your tests,  and at this point I would think manager implementation is pretty robust, but as I mentioned before, exploratory testing and adding edge conditions could uncover early/hidden issues and shield your logic for changes. 

For example, just in order to develop a little better what I'm trying to express here, with a test like the following, and keeping the focus on testing `getPregnantTreatmentType()` method,  you would catch issues in equals method, the  absent of a index in the data model allowing duplicate records, problems about the order of the expected results, etc:

```
	@Test
	public void testMgrGetPregnantTreatmentType() throws Exception {
		String code = _setupTestPregnantTreatmentType(false);
		String code2 = _setupTestPregnantTreatmentType(false);
		PregnantTreatmentType foundPregnantTreatmentType = pregnantTreatmentTypeIoOperationRepository.findOne(code);
		PregnantTreatmentType pregnantTreatmentType2 = new PregnantTreatmentType("AA", "AA description");
		pregnantTreatmentTypeIoOperationRepository.saveAndFlush(pregnantTreatmentType2);
		PregnantTreatmentType pregnantTreatmentType3 = new PregnantTreatmentType("BB", "BB description");
		pregnantTreatmentTypeIoOperationRepository.saveAndFlush(pregnantTreatmentType3);
		
		PregnantTreatmentType pregnantTreatmentType3b = new PregnantTreatmentType("BB", "BB description");
		pregnantTreatmentTypeIoOperationRepository.saveAndFlush(pregnantTreatmentType3b);

		ArrayList<PregnantTreatmentType> pregnantTreatmentTypes = pregnantTreatmentTypeBrowserManager.getPregnantTreatmentType();
		assertThat(pregnantTreatmentTypes).hasSize(3);
		
		PregnantTreatmentType pregnantTreatmentType4 = new PregnantTreatmentType("AA1", "BB description");
		pregnantTreatmentTypeIoOperationRepository.saveAndFlush(pregnantTreatmentType4);
		PregnantTreatmentType pregnantTreatmentType5 = new PregnantTreatmentType("BB1", "AA description");
		pregnantTreatmentTypeIoOperationRepository.saveAndFlush(pregnantTreatmentType5);
		
		pregnantTreatmentTypes = pregnantTreatmentTypeBrowserManager.getPregnantTreatmentType();
		assertThat(pregnantTreatmentTypes).hasSize(5);
		
		
		String description = "AA";
		
		for (PregnantTreatmentType pregnantTreatmentType : pregnantTreatmentTypes) {
			if (pregnantTreatmentType.getCode().equals(code)) {
				assertThat(pregnantTreatmentType.getDescription()).isEqualTo(foundPregnantTreatmentType.getDescription());
			} else {
				assertThat(pregnantTreatmentType.getDescription()).isNotEqualTo(foundPregnantTreatmentType.getDescription());
			}
			
			assertThat(description).isLessThanOrEqualTo(pregnantTreatmentType.getDescription());
			description = pregnantTreatmentType.getDescription();
		}
	}
```

Reply:
BTW this is already resolved!!

Reply:
Thank you guys!